lstchain/tools/lstchain_create_dl3_file.py

```diff
@@ -159,16 +154,45 @@ def setup(self):
 
         self.log.debug(f"Output DL3 file: {self.output_file}")
 
+        try:
+            QTable.read(self.input_irf, hdu=1).meta["GH_CUT"]
+            self.energy_dependent_cuts = False
+        except KeyError:
+            try:
+                QTable.read(self.input_irf, hdu="GH_CUTS")
+                self.energy_dependent_cuts = True
+            except KeyError:
+                raise ToolConfigurationError(
+                    f"{self.input_irf} does not have GH CUTS HDU, "
+                    "the energy-dependent gammaness cuts HDU, or "
+                    " any global cut information in the Header value"
+                )
+
     def start(self):
 
         self.data = read_data_dl2_to_QTable(str(self.input_dl2))
-        ## To reduce the table columns further, add a selection of columns to be read and used.
         self.effective_time, self.elapsed_time = get_effective_time(self.data)
         self.run_number = run_info_from_filename(self.input_dl2)[1]
 
         self.data = self.event_sel.filter_cut(self.data)
-        self.data = self.fixed_cuts.gh_cut(self.data)
-        self.data = add_icrs_position_params(self.data, self.source_pos)
+
+        if self.energy_dependent_cuts:
```

Comment:
This applies both energy dependent theta and gh cuts, although only one of the two cuts coul be energy dependent.